pallets/pallet-did-lookup/src/lib.rs

```diff
@@ -294,31 +297,35 @@ pub mod pallet {
 			Self::remove_association(account)
 		}
 
-		/// Transfer the storage deposit from one account to another.
+		/// Changes the deposit owner.
 		///
-		/// If the currently required deposit is different, the new deposit
-		/// value will be reserved.
+		/// The balance that is reserved by the current deposit owner will be
+		/// freed and balance of the new deposit owner will get reserved.
 		///
 		/// The subject of the call must be linked to the account.
 		/// The sender of the call will be the new deposit owner.
-		#[pallet::weight(<T as Config>::WeightInfo::transfer_deposit())]
-		pub fn transfer_deposit(origin: OriginFor<T>, account: LinkableAccountId) -> DispatchResult {
+		#[pallet::weight(<T as Config>::WeightInfo::change_deposit_owner())]
+		pub fn change_deposit_owner(origin: OriginFor<T>, account: LinkableAccountId) -> DispatchResult {
 			let source = <T as Config>::EnsureOrigin::ensure_origin(origin)?;
+			let subject = source.subject();
 
 			let record = ConnectedDids::<T>::get(&account).ok_or(Error::<T>::AssociationNotFound)?;
-			ensure!(record.did == source.subject(), Error::<T>::NotAuthorized);
+			ensure!(record.did == subject, Error::<T>::NotAuthorized);
```

Comment:
Food for thought for a potential later PR: Every pallet has their own version of `AccessDenied`. Maybe we want to normalize this in the future.

Reply:
yes agree 100%